R/fwrite.R

```diff
@@ -54,6 +56,7 @@ fwrite <- function(x, file="", append=FALSE, quote="auto",
   file <- path.expand(file)  # "~/foo/bar"
   if (append && missing(col.names) && (file=="" || file.exists(file)))
     col.names = FALSE  # test 1658.16 checks this
+  if (with_bom && !col.names) stop("with_bom can be TRUE only if col.names is TRUE")
```

Comment:
Maybe just ignore `with_bom` here (possibly with `warning`)?

Reply:
Shouldn't `with_bom && append` be a warning/error/ignore case as well?

Reply:
What BOM couldn't be used without col names? I recall using some excel data without col names.

Reply:
Because of the organization of fwrite : rows writing is threaded and every thread has its own buffer. It's difficult to write BOM in this section of fwrite. The header is written with its own buffer : with_bom uses this buffer and so col.names must be TRUE.

Reply:
> Shouldn't `with_bom && append` be a warning/error/ignore case as well?

You're right. I propose that when appending, `bom` is silently set to FALSE, like `col.names`.
See f8775587. So stop is only when creating and col.names = FALSE.